source/fundamentals/crud/write-operations/delete.txt

```diff
@@ -0,0 +1,210 @@
+delete one document.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: delete-one-async
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var result = await _restaurantsCollection.DeleteOneAsync(filter);
+
+   .. tab:: Synchronous
+      :tabid: delete-one-sync
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var result = _restaurantsCollection.DeleteOne(filter);
+
+Delete Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following code shows how you can use the asynchronous
+``DeleteManyAsync()`` method or the synchronous ``DeleteMany()`` method to
+delete all matched documents.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: delete-many-async
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var result = await _restaurantsCollection.DeleteManyAsync(filter);
+
+   .. tab:: Synchronous
+      :tabid: delete-many-sync
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var result = _restaurantsCollection.DeleteMany(filter);
+
+Find runnable examples using these methods under :ref:`additional
+information <additional-info>`.
+
+Parameters
+~~~~~~~~~~
+
+The ``DeleteOne()`` and ``DeleteMany()`` methods expect you to pass a
+*query filter* specifying which documents to match. The query filter
+determines which records are selected for deletion based on the criteria
+in :manual:`the query filter document</core/document/#query-filter-documents>`.
+
+Both methods optionally take a ``DeleteOptions`` type as an additional parameter,
+which represents options you can use to configure the delete operation.
+If you don't specify any ``DeleteOptions`` properties, the driver does
+not customize the delete operation.
+
+The ``DeleteOptions`` type allows you to configure options with the
+following properties:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``Collation``
+     - | Gets or sets the type of language collation to use when sorting results.
+
+   * - ``Comment``
+     - | Gets or sets the comment for the operation.
+
+   * - ``Hint``
+     - | Gets or sets the index to use to scan for documents.
+
+   * - ``Let``
+     - | Gets or sets the let document.
+
+Example
+~~~~~~~
+
+The following code uses the ``DeleteMany()`` method to search on the
+"borough_1" index and delete all documents whose ``address.street``
+field value includes the phrase "Pearl Street":
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var filter = Builders<Restaurant>.Filter
+          .Regex("address.street", "Pearl Street");
+
+      DeleteOptions opts = new DeleteOptions { Hint = "borough_1" };
+
+      WriteLine("Deleting documents...");
+      var result = _restaurantsCollection.DeleteMany(filter, opts);
+      
+      WriteLine($"Deleted documents: {result.DeletedCount}");
+      WriteLine($"Result acknowledged? {result.IsAcknowledged}");
```

Comment:
issue: can this be moved to source from a formatted .cs file?

Reply:
according to the [wiki page for the IO block](https://wiki.corp.mongodb.com/display/DE/How-To%3A+Add+Code+Output+for+Code+Snippets), I can pass a filepath to the [delete.cs file](https://github.com/mongodb/docs-csharp/blob/98d2f01d54a257e5fd826bef620486a0c32ac7df/source/includes/fundamentals/code-examples/crud/delete.cs) that I did create, but it doesn't seem like I can select the lines to include like I can with a literalinclude. So I just pasted the relevant code directly into the block. Let me know if you think I should just create a .cs file with just this code to include.

Reply:
I think it's okay in this instance since it's so short.